src/lua/compat.lua

```diff
@@ -0,0 +1,315 @@
+-- compat.lua -- internal module intended to solve compatibility problems in
+-- different parts of Tarantool. Introduced in gh-7000, see also gh-6912.
+
+local NEW = true
+local OLD = false
+
+local internals = require('compat_internals')
+
+local options_format = {
+    brief    = 'string',
+    obsolete = 'boolean',
+}
+
+local JSON_ESCAPE_BRIEF = [[
+Whether to escape the forward slash symbol '/' using a backslash in a json.encode()
+result. The old and the new behaviour produce a result, which is compatible
+with the JSON specification. However most of other JSON encoders don't escape
+the forward slash, so we consider the new behaviour as more safe one.
+https://github.com/tarantool/tarantool/wiki/compat%3Ajson_escape_forward_slash
+]]
+
+-- Contains static options descriptions in format specified by `options_format`.
+local options = {
+    json_escape_forward_slash = {
+        default = 'old',
+        obsolete = false,
+        brief = JSON_ESCAPE_BRIEF,
+    },
+}
+
+-- Contains postaction functions for each non obsolete option from `options`.
+local postaction = {
+    json_escape_forward_slash = function(value)
+        value = not value
+        require('json').cfg{encode_esc_slash = value}
```

Comment:
This will affect the global json instance that is returned with `require 'json'`. The function in `serializer.c` affects only new serializers that are created with `json.new()` after option is selected.

Reply:
I can approve Gleb's answer. Closing the discussion.